nubis/puppet/fluentd.pp

```diff
@@ -4,17 +4,25 @@
 }
 
 # Make fluentd run as root so it can read all log files
-file { "/etc/sysconfig":
-  ensure => "directory",
-  owner  => "root",
-  group  => "root",
-  mode    => '0755',
-}->
-file { "/etc/sysconfig/td-agent":
-  ensure => "present",
-  owner  => "root",
-  group  => "root",
-  source => "puppet:///nubis/files/fluentd.sysconfig",
+if $osfamily == 'RedHat' {
+  file { "/etc/sysconfig":
+    ensure => "directory",
+    owner  => "root",
+    group  => "root",
+    mode    => '0755',
+  }->
+  file { "/etc/sysconfig/td-agent":
+    ensure => "present",
+    owner  => "root",
+    group  => "root",
+    source => "puppet:///nubis/files/fluentd.sysconfig",
+  }
+}
+elsif $osfamily == 'Debian' {
+  exec { "change-fluentd-user":
+    command => "/usr/bin/perl -pi -e's/^(USER|GROUP)=td-agent/\$1=root/g' /etc/init.d/td-agent",
+    require => Class['Fluentd::Packages'],
+  }
```

Comment:
Yeah, and I suspect we could sprinkle stuff like that in a lot of place. Currently, we assume RedHat/Debian pretty heavily in a lot of places. Good catch!